src/lib/dapp/DAppSigner.ts

```diff
@@ -32,18 +33,53 @@ import {
 } from '@hashgraph/sdk'
 import type { ISignClient } from '@walletconnect/types'
 
-import { ledgerIdToCAIPChainId } from '../shared'
+import {
+  ExecuteTransactionResult,
+  HederaJsonRpcMethod,
+  SignAndExecuteQueryResult,
+  SignAndExecuteTransactionResult,
+  SignMessageResult,
+  SignTransactionResult,
+  base64StringToSignatureMap,
+  ledgerIdToCAIPChainId,
+  transactionBodyToBase64String,
+  transactionToBase64String,
+  transactionToTransactionBody,
+} from '../shared'
 
 export class DAppSigner implements Signer {
+  static signClient: ISignClient
+  public signerAccountId: string
+
+  /**
+   * These are the node accounts that will be utilized for signing transactions.
+   */
+  public nodesAccountIds: AccountId[] = []
+
   constructor(
     private readonly accountId: AccountId,
-    private readonly signClient: ISignClient,
     public readonly topic: string,
     private readonly ledgerId: LedgerId = LedgerId.MAINNET,
-  ) {}
+  ) {
+    this.signerAccountId = `${ledgerIdToCAIPChainId(ledgerId)}:${accountId.toString()}`
+    this.nodesAccountIds = [AccountId.fromString('0.0.3')]
+  }
+
+  static initialize(signClient: ISignClient) {
+    this.signClient = signClient
+  }
+
+  /**
+   * Set the nodes to sign transactions that require a single signature.
+   *
+   * @param {AccountId[]} nodesAccountIds
+   */
+  setNodes(nodesAccountIds: AccountId[]): void {
+    this.nodesAccountIds = nodesAccountIds
```

Comment:
Same comment. I think nodeAccountIds can be fetched more reliably using a `Client` or `Provider` instead of allowing the dApp to explicitly set them like this.

If the dApp wants to explicitly set nodeAccountIds, it should do so on the transaction itself instead of DAppSigner.

Reply:
Same suggestion :)